### PR TITLE
liquibase in spring boot

### PR DESCRIPTION
**Pull request by jdd60a**

### CODE REVIEW

File: liquibase/run-liquibase.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+#This script updates the DB schema for the user service.  It should be called before the service is started,
+#and if it has a non-zero exit code, the service *SHOULD NOT BE STARTED*. 
+#It works by:
+# 1) pulling down schema-modifying credentials from consul (see https://github.com/Victorious/git-consul-repo)
+# 2) running `liquibase migrate` to apply changes
```
Comment (jdd60a) [inline liquibase/run-liquibase.sh]: Any objections to this approach?  This script will basically be the same for all services, wanted to get some feedback before I went too much farther down this route.

Comment (gtvictorious) [reply liquibase/run-liquibase.sh]: localhost:8500 wont work in containers. you have to get the ip similar to startup.sh

Comment (gtvictorious) [reply liquibase/run-liquibase.sh]: how will liquibase know to only do this once?

Comment (jdd60a) [reply liquibase/run-liquibase.sh]: Oh good catch on the localhost: side-effect of local testing.

Liquibase knows to do this only once by managing a separate set of schema version tables that let it do an intelligent diff - see [here](http://www.liquibase.org/documentation/databasechangelog_table.html) for a little more detail

Comment (gtvictorious) [reply liquibase/run-liquibase.sh]: also, to get this to work in local-backend you have to do some trickery as well fyi.

File: pom.xml [hunk 2]
```diff
@@ -1,175 +1,178 @@
...
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>mysql</groupId>
+			<artifactId>mysql-connector-java</artifactId>
+			<scope>runtime</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.liquibase</groupId>
```
Comment (jdd60a) [inline pom.xml]: sorry for the whitespace noise, adding `liquibase-core` was the only change.  [Better Link](https://github.com/Victorious/user-service/pull/20/files?w=1)

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -27,7 +29,7 @@ spring.datasource.min-idle=10
 
 #JPA
 spring.jpa.hibernate.naming-strategy=org.hibernate.cfg.DefaultComponentSafeNamingStrategy
-spring.jpa.hibernate.ddl-auto=update
+spring.jpa.hibernate.ddl-auto=none
```
Comment (jdd60a) [inline src/main/resources/application.properties]: I'm not sure I like moving (keeping) schema management into the application layer
Get to keep the benefits of our existing pipeline (can change credentials for liquibase to be a schema, not data, account)
JPA will bail out in a consistent way (at bean init time) across all environments if liquibase schema can't be updated with latest changelog *or* doesn't match java code (in unit tests, even)

File: src/main/java/com/victorious/user/model/User.java [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
 @Entity
 @Table(name = "user",
-        indexes = {@Index(name = "user_email", columnList = "email")})
+        indexes = {@Index(name = "user-email-ix", columnList = "email")})
```
Comment (jdd60a) [inline src/main/java/com/victorious/user/model/User.java]: @gtvictorious the index name doesn't actually matter, but I wanted to keep in in sync with liquibase.

**Pull request merged**